### PR TITLE
[Merged by Bors] - perf(Convex/Between): speed up two slow refines

### PR DESCRIPTION
**Pull request by grunweg**
Previously, each refine took more than 6s in CI. Extract the last goal as a `have` as use `apply` instead, which is near-instant.
Note that `exact` or `refine` is still slow.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Convex/Between.lean [hunk 1]
```diff
@@ -153,14 +153,14 @@ theorem Function.Injective.sbtw_map_iff {x y z : P} {f : P →ᵃ[R] P'} (hf : F
 @[simp]
 theorem AffineEquiv.wbtw_map_iff {x y z : P} (f : P ≃ᵃ[R] P') :
     Wbtw R (f x) (f y) (f z) ↔ Wbtw R x y z := by
-  refine Function.Injective.wbtw_map_iff (?_ : Function.Injective f.toAffineMap)
-  exact f.injective
+  have : Function.Injective f.toAffineMap := f.injective
+  apply Function.Injective.wbtw_map_iff this
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Convex/Between.lean:157]:
```suggestion
  exact this.wbtw_map_iff
```

File: Mathlib/Analysis/Convex/Between.lean [hunk 2]
```diff
@@ -153,14 +153,14 @@ theorem Function.Injective.sbtw_map_iff {x y z : P} {f : P →ᵃ[R] P'} (hf : F
...
 
 @[simp]
 theorem AffineEquiv.sbtw_map_iff {x y z : P} (f : P ≃ᵃ[R] P') :
     Sbtw R (f x) (f y) (f z) ↔ Sbtw R x y z := by
-  refine Function.Injective.sbtw_map_iff (?_ : Function.Injective f.toAffineMap)
-  exact f.injective
+  have : Function.Injective f.toAffineMap := f.injective
+  apply Function.Injective.sbtw_map_iff this
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Convex/Between.lean:163]:
```suggestion
  exact this.sbtw_map_iff
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks!

File: Mathlib/Analysis/Convex/Between.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Analysis/Convex/Between.lean:157]: Careful: using dot notation works - using `exact` or `refine` is slow. `apply` is fast. I've switched to dot notation, but stayed with `apply`.

**Comment by YaelDillies** [comment]
Can we put adaptation notes or something? Else someone will come and regolf the proofs

File: Mathlib/Analysis/Convex/Between.lean [hunk 3]
```diff
@@ -153,14 +153,14 @@ theorem Function.Injective.sbtw_map_iff {x y z : P} {f : P →ᵃ[R] P'} (hf : F
 @[simp]
 theorem AffineEquiv.wbtw_map_iff {x y z : P} (f : P ≃ᵃ[R] P') :
     Wbtw R (f x) (f y) (f z) ↔ Wbtw R x y z := by
-  refine Function.Injective.wbtw_map_iff (?_ : Function.Injective f.toAffineMap)
-  exact f.injective
+  have : Function.Injective f.toAffineMap := f.injective
+  apply this.wbtw_map_iff
```
Comment (grunweg) [inline Mathlib/Analysis/Convex/Between.lean:157]:
```suggestion
  -- `refine` or `exact` are very slow, `apply` is fast. Please check before golfing.
  apply this.wbtw_map_iff

File: Mathlib/Analysis/Convex/Between.lean [hunk 4]
```diff
@@ -153,14 +153,14 @@ theorem Function.Injective.sbtw_map_iff {x y z : P} {f : P →ᵃ[R] P'} (hf : F
...
 
 @[simp]
 theorem AffineEquiv.sbtw_map_iff {x y z : P} (f : P ≃ᵃ[R] P') :
     Sbtw R (f x) (f y) (f z) ↔ Sbtw R x y z := by
-  refine Function.Injective.sbtw_map_iff (?_ : Function.Injective f.toAffineMap)
-  exact f.injective
+  have : Function.Injective f.toAffineMap := f.injective
+  apply this.sbtw_map_iff
```
Comment (grunweg) [inline Mathlib/Analysis/Convex/Between.lean:163]:
```suggestion
  -- `refine` or `exact` are very slow, `apply` is fast. Please check before golfing.
  apply this.sbtw_map_iff

File: Mathlib/Analysis/Convex/Between.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Analysis/Convex/Between.lean]: Hmm, it was not noticeably slow when I tried it - certainly nowhere near what it was before. How much of a difference do you see?

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**